netbox/utilities/jinja2.py

```diff
@@ -29,10 +29,16 @@ def get_source(self, environment, template):
 
         # Find and pre-fetch referenced templates
         if referenced_templates := find_referenced_templates(environment.parse(template_source)):
```

Comment:
There's a very odd little bug hidden here when not using dynamic resolution for related templates. We need to cast the generator returned by `find_referenced_templates()` to an iterable to avoid exhausting it on the check for None. (Otherwise it ends up empty when passed as the value for `path__in`.)
```suggestion
        if referenced_templates := tuple(find_referenced_templates(environment.parse(template_source))):
```

Reply:
Nice catch!